go/vt/vttablet/tabletmanager/tm_init.go

```diff
@@ -730,10 +730,18 @@ func (tm *TabletManager) checkMysql() error {
 	return nil
 }
 
+const portCheckTimeout = 5 * time.Second
+
+func (tm *TabletManager) getMysqlPort() (int32, error) {
```

Comment:
It seems overkill to define a function that is used exactly once. Why can't all this be folded into findMysqlPort? It will grow from 10 lines to 15 lines.

Reply:
@deepthi Because it's setting up a context and you can't defer that properly in a `for` loop. So hence the separate function so it's not needed to manually cancel that which is more error prone.